src/Adapter/DefaultConfigurator.php

```diff
@@ -185,4 +190,34 @@ public function validateUrl($url)
 
         return $url;
     }
+
+    /**
+     * Get the base adapter name.
+     *
+     * @return string
+     */
+    public function getBaseAdapterName()
+    {
+        return $this->baseAdapterName;
+    }
+
+    /**
+     * Set the base adapter name.
+     *
+     * @return string
+     */
+    public function setBaseAdapterName($name)
```

Comment:
This can be moved to the constructor as a configurator should never change its actual adapter base-name.
Setters to require something is set (without it the class will not work) will the make the class more prone to bugs.

Besides that :+1: for this change.